### ISSUE TITLE
Drop-down menues in the overview not lockable and/or permanently open

### ISSUE THREAD

**Issue by schnudd31do3** [znuny/Znuny#503]
### Environment

- OS: Debian 12
- Browser: Chrome
- Znuny version: 7.0.12

### Expected behaviour

The drop down menues in the overview on the left side should be able to be opened and closed. They should be closed by default, otherwise they partially overlap the list of tickets.

### Actual behaviour

The drop down menues are permanently open and can not be closed.
![image](https://github.com/znuny/Znuny/assets/72391822/fae1e532-64b6-4df0-9d3d-4f67481b89fb)
![image](https://github.com/znuny/Znuny/assets/72391822/359acc03-af3e-4578-b4d3-77501a13eb4e)

**Comment by mo-azfar** [comment]
Same as mine..
Can show show the console log?

Mine

![image.png](https://github.com/znuny/Znuny/assets/62957738/e1796358-1308-49bc-9929-f1a1a32b02de)

**Comment by schnudd31do3** [comment]
Looks similar to mine:
![image](https://github.com/znuny/Znuny/assets/72391822/55c6c8f3-fe57-4b88-a0c8-22b62b2208c8)

**Comment by hanneshal** [comment]
Hello @mo-azfar @schnudd31do3 
could one of you disable the Sysconfig: Loader::Enabled::JS and check the console log again.

This will print a readable error.

Thanks

**Comment by mo-azfar** [comment]
> Hello @mo-azfar @schnudd31do3 could one of you disable the Sysconfig: Loader::Enabled::JS and check the console log again.
> 
> This will print a readable error.
> 
> Thanks

The moment Loader::Enabled::JS is disabled, the console log error gone

![image](https://github.com/znuny/Znuny/assets/62957738/355f1e2f-5f73-4f4b-8989-69144b0f85ef)

**Comment by mo-azfar** [comment]
I do some checking previously, 

just compared some js files..
the are called at Dashboard and Statistic module..

7.0.11 use thirdparty/d3-3.5.7/d3.min.js
7.0.12 use thirdparty/d3-3.5.7/d3.js

restoring d3.min.js from 7.0.11 to 7.0.12 and set back the config to point to d3.min.js and the error gone
and dashboard/statistic working again..

**Comment by schnudd31do3** [comment]
> > Hello @mo-azfar @schnudd31do3 could one of you disable the Sysconfig: Loader::Enabled::JS and check the console log again.
> > This will print a readable error.
> > Thanks
> 
> The moment Loader::Enabled::JS is disabled, the console log error gone
> 
> ![image](https://user-images.githubusercontent.com/62957738/276847601-355f1e2f-5f73-4f4b-8989-69144b0f85ef.png)

This did not make any difference. Means, did not help.

**Comment by schnudd31do3** [comment]
Where can I set the config for d3.min.js?

I copied the ./thirdparty/d3-3.5.7/d3.min.js from 7.0.11 to 7.0.12 and renamed it to d3.js -> the drop-down menues have gone. This is no solution

**Comment by hanneshal** [comment]
Hi, 
Question. 
Does the d3.js acutally exist on your instances? Is there a 404 not found in the network tab?

If there is no error and the file exists, can you please run:

`/opt/znuny/bin/znuny.Console.pl Maint::Config::Rebuild --cleanup`

If you modified the default loader, you need to reset the loader config yourself
`Loader::module::AgentDashboard###001-Framework`
and
`Loader::module::AgentStatistics###001-Framework`

Regards
Johannes

**Comment by schnudd31do3** [comment]
- "Does the d3.js acutally exist on your instances?" -> yes

- "Is there a 404 not found in the network tab?" -> no

![image](https://github.com/znuny/Znuny/assets/72391822/3ed5d80e-fa18-49b5-bf5d-f4a70d94177e)

- "/opt/znuny/bin/znuny.Console.pl Maint::Config::Rebuild --cleanup" -> no difference

- There are two js files. Which one has to be changed?

![image](https://github.com/znuny/Znuny/assets/72391822/3a33fce5-2829-4a18-bf11-ffdf9e2289e6)

- Changing the d3-config did not solve the problem, because this solution only disables the drop-down menues(see above).

**Comment by hanneshal** [comment]
Both settings are correct, no need to change it.
The JS Error is gone, which was the point.

If the menu still does not open, we can carry on with the bug search. 
@mo-azfar wrote:
```
restoring d3.min.js from 7.0.11 to 7.0.12 and set back the config to point to d3.min.js and the error gone
and dashboard/statistic working again..
```

So I think he took over the issue for the DashboardStats, and not with the menu problem itself.

**Comment by schnudd31do3** [comment]
"The JS Error is gone, which was the point." unfortunately not. It only causes that the menue is not shown any more. Ok in this case we could say the error has gone by making the menue invisible (in my case).

**Comment by mo-azfar** [comment]
1. d3.js is exist.

2. Both of the default setting as below correctly point to d3.js
`Loader::module::AgentDashboard###001-Framework`
`Loader::module::AgentStatistics###001-Framework`

3. Error in console log as described in 1st post
Also effected some display / filter button at the Dashboard and Statistic.

4. executed `/opt/znuny/bin/znuny.Console.pl Maint::Config::Rebuild --cleanup`, still the issue occurred.

5. Since this is working on version 7.0.11 and below, i took a deeper look at the `Loader::module::AgentDashboard###001-Framework` and  `Loader::module::AgentStatistics###001-Framework`

version 7.0.11 used `thirdparty/d3-3.5.7/d3.min.js`
version 7.0.12 use `thirdparty/d3-3.5.7/d3.js`


6.  I restoring d3.min.js (from 7.0.11) to 7.0.12 and set back the config to point to d3.min.js and the error gone
and dashboard/statistic working again

    `Loader::module::AgentDashboard###001-Framework` and  
    `Loader::module::AgentStatistics###001-Framework`

![image](https://github.com/znuny/Znuny/assets/62957738/10a3ca9c-1489-4aa9-920a-01a6fdf2e58c)


7. Below is the video after changes.

https://github.com/znuny/Znuny/assets/62957738/3d4ef00b-89cc-4ccd-b43c-976bdf113a85

**Comment by schnudd31do3** [comment]
1. I changed the config as shown.
2. /opt/znuny/bin/znuny.Console.pl Maint::Config::Rebuild --cleanup
3. reboot of the server
4. Result like shown in the video from @mo-azfar 

Thanks.

**Comment by mo-azfar** [comment]
> 1. I changed the config as shown.
> 2. /opt/znuny/bin/znuny.Console.pl Maint::Config::Rebuild --cleanup
> 3. reboot of the server
> 4. Result like shown in the video from @mo-azfar
> 
> Thanks.

So, it back to normal / expectation? 
If yes, i suspect perhaps the d3.js not playing well with based code or either one

**Comment by schnudd31do3** [comment]
"So, it back to normal / expectation?" yes. In hindsight in my case I needed to restart the znuny services (or reboot of the server) to make it working.

Perhaps someone from the developers of Znuny can say why the switch to d3.js was necessary. Apparently, the software works better with the previous d3.min.js. But maybe we don't see all the effects on the software by going back to the previous d3.min.js

**Comment by mo-azfar** [comment]
> "So, it back to normal / expectation?" yes. In hindsight in my case I needed to restart the znuny services (or reboot of the server) to make it working.
> 
> Perhaps someone from the developers of Znuny can say why the switch to d3.js was necessary. Apparently, the software works better with the previous d3.min.js. But maybe we don't see all the effects on the software by going back to the previous d3.min.js

Yeah..there maybe and effect / not by going back to d3.min.js..
as this like temporary fix, i suggest keep track of this until see proper official fix.

**Comment by hanneshal** [comment]
We did some digging around in this.
The problem is related to the JS Loader / Minifier. 

1) Loader Disabled -> Works
2) JavaScript::Minifier (Old Loader) -> Works
3) JavaScript::Minifier::XS (Default) -> does not work

We need to investigate this further.
Possible workarounds, as already shown.
a) Disable Loader (not recommended)
b) Replace content from d3.js with d3.min.js
c) Uninstall JavaScript::Minifier::XS 

Regards

**Comment by constantin-vitrado** [comment]
Ok, the solutions were outlined here already, but I had difficulties following them. So my path to fixing the frontend was:
1. fetch the d3.min.js from the previous release https://download.znuny.org/releases/znuny-7.0.11.zip
2. copy it into /var/httpd/htdocs/js/thirdparty/d3-3.5.7/
3. move d3.min.js to d3.js
4. bin/znuny.Console.pl Maint::Config::Rebuild --cleanup

**Comment by jepf** [comment]
Internal MRs: 1037, 1038

**Comment by jepf** [comment]
This will be fixed in release 7.0.13 (and 6.5.5).

**Comment by peralta00** [comment]
https://github.com/znuny/Znuny/issues/518
